src/goto-programs/restrict_function_pointers.cpp

```diff
@@ -0,0 +1,415 @@
+/*******************************************************************\
+
+Module: GOTO Program Utilities
+
+Author: Diffblue Ltd.
+
+\*******************************************************************/
+
+#include "restrict_function_pointers.h"
+
+#include <ansi-c/expr2c.h>
+#include <json/json_parser.h>
+#include <util/expr_iterator.h>
+#include <util/string_utils.h>
+
+#include <algorithm>
+#include <fstream>
+#include <iostream>
+
+namespace
```

Comment:
:question: What is the purpose of this `namespace`?

Reply:
This is just to hide implementation-specific symbols, same as writing `static` in front of them. I just hide functions that don’t need to be global from the global namespace out of habit.